src/celllistimpl.h

```diff
@@ -0,0 +1,911 @@
+        return Base::isCell(coordinates + offset);
+    }
+
+    using Base::Base; // use parent constructors
+};
+
+/**
+ * @brief Periodic boundary conditions.
+ *
+ * Both spatial coordinates and cell coordinates are first converted to the original image and then evaluated.
+ * To avoid repeated cell inclusion by completing a full circle in periodic boundary conditions, offsets are limited
+ * in both directions assuming symetry.
+ *
+ * @todo Currently only full or none PBC is supperted. To support geometries like a slit, more fine distingtion is
+ * needed.
+ *
+ * @tparam TGridType  a cell grid type declaration
+ */
+template <typename TGridType>
+class PeriodicBoundaryGrid : public GridBase<TGridType>, virtual public AbstractNeighborGrid<TGridType> {
+    using Base = GridBase<TGridType>;
+
+  public:
+    using GridType = typename AbstractNeighborGrid<TGridType>::GridType;
+    using typename GridType::CellCoord;
+    using typename GridType::CellIndex;
+    using typename GridType::Point;
+    using typename GridType::SpaceAxis;
+
+    /**
+     * @brief Compute the cell coordinates from the cell index.
+     *
+     * ⌊ (index mod cell_index_to_coordinates) / cell_coordinates_to_index ⌋
+     *
+     * @param index
+     * @return  cell coordinates
+     */
+    CellCoord coordinates(const CellIndex index) const override { return Base::coordinates(index); }
+
+    /**
+     * @brief Compute the cell index from the cell coordinates.
+     *
+     * coordinates mod cell_list_end
+     *
+     * @param coordinates  cell coordinates
+     * @return  cell index
+     */
+    CellIndex index(const CellCoord& coordinates) const override {
+        const auto pbc_coord = (coordinates - (coordinates / this->getCellListEnd()) * this->getCellListEnd()).eval();
+        return Base::index(pbc_coord);
+    }
+
+    /**
+     * @brief Compute cell coordinates from a spatial position.
+     *
+     * ⌊ position / cell_dimension ⌋
+     *
+     * @param position
+     * @return  cell coordinates
+     */
+    CellCoord coordinatesAt(const Point& position) const override {
+        assert(isCellAt(position));
+        return (position / this->getCell()).floor().template cast<CellIndex>();
+    }
+
+    /**
+     * @brief Compute the cell index from the spatial position.
+     * @param position   spatial position
+     * @return  cell index
+     */
+    CellIndex indexAt(const Point& position) const override { return this->index(this->coordinatesAt(position)); }
+
+    /**
+     * @brief Verifies if coordinates are valid in the grid.
+     *
+     * Any coordinates are valid under full PBC.
+     *
+     * @param coordinates
+     * @return  always true
+     */
+    bool isCell(const CellCoord&) const override { return true; }
+
+    /**
+     * @brief Verifies if spatial coordinates are in the box.
+     *
+     * Any coordinates are valid under full PBC.
+     *
+     * @param point
+     * @return  always true
```

Comment:
why?

Reply:
error if really outside?

Reply:
To check if the given coordinates, computed for example as cell + offset, can be translated to a cell index. That is always possible under periodic boundary conditions, but not under fixed boundary conditions, e.g., for the cells at edges. That is the sole purpose of the grid to hide boundary condition details. Documentation extended accordingly.